### PR TITLE
feat: take file upload limits from server

### PR DESCRIPTION
**Pull request by seavan**
Slava now provides the following information:

this.quota.resultingQuotas.upload[0].limit

- is null if there are no limits
- is a number if there are limits.

### CODE REVIEW

File: src/helpers/path.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+ * @param {string} expression Path expression, e.g. quota.resultingQuotas.upload[metric=maxSize,period=total].limit
+ * @returns {function} Function, which will return undefined or value if evaluation is a success
+ */
+function compile(expression) {
+    if (compileCache[expression]) return compileCache[expression];
+    let result = a => a;
+    expression.split('.').forEach(path => {
+        const prevCaller = result;
+        const found = path.match(/(.*)\[(.*?)\]/);
+        if (found) {
```
Comment (anri-asaturov) [inline src/helpers/path.js]: Add comments inside code describing what and why are we doing

File: src/helpers/path.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
+const compileCache = {};
+
```
Comment (anri-asaturov) [inline src/helpers/path.js]: this file has to be a module of its own (@module) to render correctly in dokumentationjs. Check other similar modules.

File: src/helpers/path.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+/**
+ * Creates a function which evaluates path applied to given object,
+ * step by step. The function will return 'undefined' if any exception
+ * arises during the call
+ * @param {string} expression Path expression, e.g. quota.resultingQuotas.upload[metric=maxSize,period=total].limit
+ * @returns {function} Function, which will return undefined or value if evaluation is a success
```
Comment (anri-asaturov) [inline src/helpers/path.js]: add @public or @protected

File: src/models/user/user.js [hunk 4]
```diff
@@ -416,10 +417,8 @@ class User {
      * @public
      */
     canUploadMaxFileSize = (size) => {
-        if (this.isProUser) return true;
-        const adjustedSize = this._adjustedOverheadFileSize(size);
-        if (this.isPremiumUser) return adjustedSize < config.premiumMaxSingleFileUploadSize;
-        return adjustedSize < config.basicMaxSingleFileUploadSize;
+        const limit = path('quota.resultingQuotas.upload[metric=maxSize,period=total].limit', this);
```
Comment (anri-asaturov) [inline src/models/user/user.js]: change other parts of code to use your ugly path parser to justify merging this :-D

File: src/helpers/path.js (hunk 2 above)
Comment (anri-asaturov) [inline src/helpers/path.js]: Add tests, so nobody would suffer from bugs in parser or edge cases.

Comment (anri-asaturov) [reply src/helpers/path.js]: rename path.js to something else, 'path' is usually associated with file system.
